### DISCUSSION TITLE
AxHttp API doesn't seem to work with HTTPS

### DISCUSSION THREAD

**Discussion by polodroid74** [AxisCommunications/acap3-examples#204] [Q&A]
Hi,
As an ACAP3 developer, I'm developing an application using a settings page file (specified in the package.conf file).
In order to use this setting page, I have a thread running the AxHttp API following the example of the _ax_http_serve_request.c_ file ([https://www.axis.com/techsup/developer_doc/acap3/3.4/api/axhttp/html/ax_http_serve_request_8c-example.html](https://www.axis.com/techsup/developer_doc/acap3/3.4/api/axhttp/html/ax_http_serve_request_8c-example.html))

Everything works fine when i'm connected to my application using HTTP. However, when I switch the camera to HTTPS, the webpage can not interact with the AxHTTP server anymore.

Investigating the bug, I found that I an _net::ERR_HTTP2_PROTOCOL_ERROR_ occurs when using th HTTPS protocol.

From what I understand (but I'm not an expert in web development), the web page and the server interact like this : 
![image](https://user-images.githubusercontent.com/24894685/226910154-1d2bc6ef-9e87-4c06-bf47-eaeefeb09760.png)
where the AxHttp object is the server and the CGI program some code I have written in C++.

When I'm using HTTPS I'm able to add some logs on the CGI program side. It appeared that the request arrived correctly to my CGI program but didn't succeed to "go back" to the web interface.

Again, I'm not a web developer so I probably made some mistakes (in my code and in my explainations) but I wanted to know if this was a common issue with HTTPS and AxHTTP ?

NB : I've also tested some basic AXiS CGI requests ( like this one [http://192.168.0.223/axis-cgi/admin/param.cgi?action=list&group=Brand.ProdNbr](http://192.168.0.223/axis-cgi/admin/param.cgi?action=list&group=Brand.ProdNbr) which worked both in HTTP and HTTPS? This makes me think that the problem is related to my use of the AxHTTP API.

**Comment by pataxis** [comment]
Hi @polodroid74 , thanks for your question, we'll get back to you when we have had time to look in to this.

**Reply by polodroid74** [reply]
Ok, I've figured it out.
In the code, I added manually to my HTTP Response the header `HTTP/2`
But it seems it was a mistake. Indeed, I think this is the camera responsibility to add and handle this.
Removing this line solved the problem !
